tests/submit_recovery_share.sh

```diff
@@ -0,0 +1,85 @@
+
+set -e
+
+function cleanup() {
+  rm -f member_enc_privk.raw network_enc_pubk.raw
+}
+trap cleanup 0
+
+function usage()
+{
+    echo "Usage:"""
+    echo "  $0 --rpc-address rpc_address --member-enc-privk member_enc_privk.pem --network-enc-pubk network_enc_pubk.pem [CURL_OPTIONS]"
+    echo "Retrieves the encrypted recovery share for a given member, decrypts the share and submits it for recovery."
+    echo ""
+    echo "A sufficient number of recovery shares must be submitted by members to initiate the end of recovery procedure."
+    echo "Note: Requires step CLI."
+}
+
+while [ "$1" != "" ]; do
+    case $1 in
+        -h|-\?|--help)
+            usage
+            exit 0
+            ;;
+        --rpc-address)
+            node_rpc_address="$2"
+            ;;
+        --member-enc-privk)
+            member_enc_privk="$2"
+            ;;
+        --network-enc-pubk)
+            network_enc_pubk="$2"
+            ;;
+        *)
+            break
+    esac
+    shift
+    shift
+done
+
+if ! [ -x "$(command -v step)" ]; then
+    echo "Error: step CLI is not installed on your system or not in your path."
+    echo "See https://microsoft.github.io/CCF/master/members/accept_recovery.html#submitting-recovery-shares"
+    exit 1
+fi
+
+if [ -z "$node_rpc_address" ]; then
+    echo "Error: No node RPC address in arguments (--rpc-address)"
+    exit 1
+fi
+
+if [ -z "$member_enc_privk" ]; then
+    echo "Error: No member encryption private key in arguments (--member-enc-privk)"
+    exit 1
+fi
+
+if [ -z "$network_enc_pubk" ]; then
+    echo "Error: No defunct network encryption public key in arguments (--network-enc-pubk)"
+    exit 1
+fi
+
+# Retrieve encrypted recovery share and nonce
+resp=$(curl -sS --fail -X GET https://"${node_rpc_address}"/members/recovery_share "${@}")
+encrypted_share="$(echo "${resp}" | jq -r .encrypted_recovery_share)"
+nonce="$(echo "${resp}" | jq -r .nonce)"
+
+# Parse raw private key from SubjectPublicKeyInfo DER format, as generated by keygenerator.sh --gen-enc-key
+der_header_privk_len=14
+openssl asn1parse -in "${member_enc_privk}" -strparse "${der_header_privk_len}" -out member_enc_privk.raw -noout
+
+# Parse raw public key generated by network
+der_header_pubk_len=9
+openssl asn1parse -in "${network_enc_pubk}" -i -strparse "${der_header_pubk_len}" -out network_enc_pubk.raw -noout
+
+# step CLI base64 standard to URL encoding for input
+encrypted_share_b64_url=$(echo "${encrypted_share}" | sed 's/+/-/g; s/\//_/g')
+
+# Decrypt encrypted share with nonce, member private key and defunct network public key
+decrypted_share=$(echo "${encrypted_share_b64_url}" | step crypto nacl box open base64:"${nonce}" network_enc_pubk.raw member_enc_privk.raw | openssl base64 -A)
+
+# Finally, submit decrypted share
+curl -i -sS --fail https://"${node_rpc_address}"/members/recovery_share/submit "${@}" -H "Content-Type: application/json" -d '{"recovery_share": "'"${decrypted_share}"'"}'
```

Comment:
The decrypted share would be better placed in a file created by mkstemp, than passed inline where it can be very widely read by other users.

Reply:
This is now done. For the decrypted share in particular, the `recovery_share/submit` endpoint now takes the raw shares as input so we can pipe the decryption command directly to `curl`.